src/Type/Accessory/AccessoryArrayListType.php

```diff
@@ -247,6 +247,15 @@ public function shuffleArray(): Type
 		return $this;
 	}
 
+	public function sliceArray(Type $offsetType, Type $lengthType, TrinaryLogic $preserveKeys): Type
+	{
+		if ($offsetType instanceof ConstantIntegerType && $offsetType->getValue() === 0) {
```

Comment:
Note to self: should be a super type check and support for preserveKeys no can be added